### PR TITLE
[doc][modules] update to modules documentation

### PR DESCRIPTION
**Pull request by felipemello1**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

#### Changelog
- Added docstring that were missing;
- Changed some variable names like i, k, v to more meaningful names;
- Updated RST with missing API facing functions and deleted non existing ClassifierTransfomer
- Minor changes in existing docstrings

#### Test plan

`make html` - confirmed it rendered well

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add unit tests for any new functionality
- [ ] update docstrings for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
	- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

File: docs/source/api_ref_modules.rst [hunk 1]
```diff
@@ -41,7 +42,9 @@ PEFT Components
     peft.AdapterModule
     peft.get_adapter_params
     peft.set_trainable_params
-
+    peft.validate_missing_and_unexpected_for_lora
```
Comment (joecummings) [inline docs/source/api_ref_modules.rst:45]: Can we register "validate_state_dict_for_lora" in the docstring as a function so it points there?

File: docs/source/api_ref_modules.rst [hunk 2]
```diff
@@ -41,7 +42,9 @@ PEFT Components
...
+    peft.validate_state_dict_for_lora
+    peft.disable_adapter
```
Comment (joecummings) [inline docs/source/api_ref_modules.rst:47]: The example here is not formatted correctly.

File: torchtune/modules/tokenizers/_tiktoken.py [hunk 3]
```diff
@@ -290,11 +290,11 @@ def decode(
         """
         if truncate_at_eos:
             try:
-                k = token_ids.index(self.eos_id)
+                idx_eos = token_ids.index(self.eos_id)
```
Comment (joecummings) [inline torchtune/modules/tokenizers/_tiktoken.py:293]: thank you for this :)

**Review by joecummings — COMMENTED** [review]
Couple of nits, mostly looks good.

File: torchtune/modules/tokenizers/_utils.py [hunk 4]
```diff
@@ -42,6 +42,12 @@ def truncate(
     max_seq_len: int,
     eos_id: Union[int, bool],
 ):
+    """
```
Comment (RdoubleA) [inline torchtune/modules/tokenizers/_utils.py:45]: actually, do you mind removing this function? this is not used anywhere afaik and is duplicated in `torchtune.data.truncate`. I thought I had removed this but for some reason it remains...

File: torchtune/modules/peft/peft_utils.py [hunk 5]
```diff
@@ -241,23 +241,42 @@ def get_merged_lora_ckpt(
...
+    and if a module has an 'adapter_params' attribute that is callable and a 'disabled' attribute,
+    it sets 'disabled' to True. Then, the control is given back to caller. Once that finalizes,
+    it sets 'disabled' back to False for all modules that were temporarily disabled.
+
+    Args:
+        model (nn.Module): The neural network model whose adapters are to be temporarily disabled.
+    Yields:
+        None: This function yields control back to the caller, with the adapters disabled.
+    Example:
+        with disable_adapter(model):
```
Comment (RdoubleA) [inline torchtune/modules/peft/peft_utils.py:259]: you will probably need to use `>>>` to format this code snippet correctly, see other examples in other docstrings

**Review by joecummings — APPROVED** [review]
LGTM, thanks!
